### PR TITLE
feat: introduce validate status and green check icon for input fields 

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
[FX-2502]

### Description

- TODO: changeset should be added
- add new `showValidState` prop to `FieldWrapper`
- replace `error` prop with `validateStatus` prop to both support error and success states
- add valid-icon InputAdornment for `<Input>`, `<PasswordInput>`, `<NumberInput>` and `<TagSelector>`
- add `aria-invalid=false` to all inputs indirectly (which caused to update all input jest snapshots)
- add or update related tests

### How to test

- FIXME: Add the steps describing how to verify your changes

### Screenshots
Screen Shot 2022-03-11 at 11 04 24

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-2502]: https://toptal-core.atlassian.net/browse/FX-2502?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx [hunk 1]
```diff
@@ -61,31 +62,32 @@ type FieldMeta<T> = FieldMetaState<T> & {
   dirtyAfterBlur?: boolean
 }
 
-const getInputError = <T extends ValueType>(
+const getInputValidateStatus = <T extends ValueType>(
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:65]: The result type for this function can vary. It can even be split into two different functions. My purpose was configuring the validateStatus and returning the correct error with the same logic.

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx [hunk 2]
```diff
@@ -61,31 +62,32 @@ type FieldMeta<T> = FieldMetaState<T> & {
...
   meta: FieldMeta<T>,
-  formConfig: FormConfigProps
-) => {
+  formConfig: FormConfigProps,
+  showValidState?: boolean
+): [string | undefined, ValidateStatus | undefined] => {
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:69]: It was null before, I don't see any difference. If someone has better reasoning, I can revert them to null's

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx [hunk 3]
```diff
@@ -116,7 +118,7 @@ const getProps = ({
   }
 
   return {
-    error: Boolean(error)
+    validateStatus
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:121]: It wasn't an optional boolean before, now it is. Because in the initial render, there is no way to say that field is in error or success state.

File: packages/picasso-forms/src/RadioGroup/RadioGroup.tsx [hunk 4]
```diff
@@ -19,7 +19,7 @@ export const RadioGroup = (props: Props) => {
       <FieldWrapper {...rest} type='radio'>
         {radioGroupProps => {
           // eslint-disable-next-line @typescript-eslint/no-unused-vars
-          const { error, ...restRadioGroupProps } = radioGroupProps
+          const { validateStatus, ...restRadioGroupProps } = radioGroupProps
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/RadioGroup/RadioGroup.tsx:22]: The same reason as before, avoiding console errors about not passing the prop to HTML.

File: packages/picasso-forms/src/Rating/Rating.tsx [hunk 5]
```diff
@@ -13,7 +13,12 @@ export type RatingStarsProps = PicassoRatingStarsProps &
 
 const Stars = (props: RatingStarsProps) => (
   <FieldWrapper<PicassoRatingStarsProps> {...props} type='number'>
-    {inputProps => <PicassoRating.Stars {...inputProps} />}
+    {(inputProps: RatingStarsProps) => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { validateStatus, ...rest } = inputProps
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/Rating/Rating.tsx:18]: Same as RadioGroup, the error state is not being by Rating components.

File: packages/picasso/src/Autocomplete/Autocomplete.tsx [hunk 6]
```diff
@@ -277,7 +291,7 @@ export const Autocomplete = forwardRef<HTMLInputElement, Props>(
           <InputComponent
             {...rest}
             {...getInputProps()}
-            error={error}
+            validateStatus={validateStatus || (error ? 'error' : undefined)}
```
Comment (ozgurkececioglu) [inline packages/picasso/src/Autocomplete/Autocomplete.tsx:294]: Temporary solution until we decide to get rid of `error` prop.

File: packages/picasso/src/Autocomplete/__snapshots__/test.tsx.snap [hunk 7]
```diff
@@ -212,6 +212,7 @@ exports[`Autocomplete static behavior renders 1`] = `
         >
           <input
             aria-autocomplete="list"
+            aria-invalid="false"
```
Comment (ozgurkececioglu) [inline packages/picasso/src/Autocomplete/__snapshots__/test.tsx.snap:215]: For some input types, we were passing not optional boolean for `error` prop. Now, I applied it to all of them to make it more consistent. I think there is no harm to pass this attribute into HTML.

File: packages/picasso/src/FileInput/FileInput.tsx [hunk 8]
```diff
@@ -60,7 +60,7 @@ export const FileInput = forwardRef<HTMLInputElement, Props>(function FileInput(
       <Button
         size='small'
         variant='secondary'
-        disabled={disabled || preventAddingNewFiles}
+        disabled={Boolean(disabled || preventAddingNewFiles)}
```
Comment (ozgurkececioglu) [inline packages/picasso/src/FileInput/FileInput.tsx:63]: Unrelated change. It was showing an error about this in the console.

File: packages/picasso/src/Input/Input.tsx [hunk 9]
```diff
@@ -246,6 +254,16 @@ const EndAdornment = (props: EndAdornmentProps) => {
     )
   }
 
+  if (validateStatus === 'success') {
```
Comment (ozgurkececioglu) [inline packages/picasso/src/Input/Input.tsx:257]: Since `<Input>` component is being used by many others, I didn't introduce a complex logic here, because the current perspective about showing green check is fulfilled. If we need it for like `DatePicker` or `TimePicker`, this won't be working. We can figure it out later when the time comes.

File: packages/shared/src/types.ts [hunk 10]
```diff
@@ -97,3 +97,5 @@ export interface TransitionProps {
   onExited?: (node: HTMLElement) => void
   timeout?: number | { enter?: number; exit?: number; appear?: number }
 }
+
+export type ValidateStatus = 'success' | 'error'
```
Comment (ozgurkececioglu) [inline packages/shared/src/types.ts:101]: I prefer keeping it here rather than direct usage inside the input files because when we decide to add a new state here, it would be hard to refactor everywhere

**Review by TomasSlama — COMMENTED** [review]
Good job @ozgurkececioglu, it looks cool.  I left some questions and I will add one more here. Do we not support selects?

**Comment by ozgurkececioglu** [comment]
> Good job @ozgurkececioglu, it looks cool. I left some questions and I will add one more here. Do we not support selects?

@TomasSlama according to [my question](https://toptal-core.slack.com/archives/CCC3GP6CC/p1646827199085899) and answer, we only need this for the components handled in this PR. Personally, it doesn't make much sense for Select because there is no value like  `middle` for select, user may select an item or not. So, it is hard to define a moment to show green check.

File: packages/picasso-forms/src/Form/story/ValidateStatus.example.tsx [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+  const [skillInputValue, setSkillInputValue] = useState<string>(
+    EMPTY_INPUT_VALUE
+  )
+  const skillOptions = filterOptions(skillInputValue, skills)
+
+  return (
+    <Form
+      autoComplete='off'
+      onSubmit={values => window.alert(values)}
+      initialValues={{ gender: 'female' }}
```
[the remark this answers is not in the document]
Comment (ozgurkececioglu) [reply packages/picasso-forms/src/Form/story/ValidateStatus.example.tsx:36]: If we add a new prop to `<Form>`, it would be hard to read it from fields. So, I would prefer using `<Form.ConfigProvider>` for this purpose. WDYT?

[the remark this answers is not in the document]
Comment (TomasSlama) [reply packages/picasso-forms/src/Form/story/ValidateStatus.example.tsx:36]: Oh, I didn't know we have a such provider, I think it makes sense 👍

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx (hunk 2 above)
Comment (dmaklygin) [reply packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:69]: it's kind of unclear, what should be in the first index of the array and as well in the second.
WDYT if we convert it to a special type, like: `InputValidationStatus`:
```
type InputValidationStatus = {
  error?: string
  status?: ValidateStatus
}
```

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx [hunk 12]
```diff
@@ -61,31 +62,32 @@ type FieldMeta<T> = FieldMetaState<T> & {
...
-  }
-
-  if (!meta.error && !meta.submitError) {
-    return null
+    return [undefined, undefined]
   }
 
   if (!meta.touched) {
-    return null
+    return [undefined, undefined]
```
Comment (dmaklygin) [inline packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:75]: ...then, you might return only an empty object

File: packages/picasso/src/Autocomplete/Autocomplete.tsx [hunk 13]
```diff
@@ -159,6 +164,15 @@ export const Autocomplete = forwardRef<HTMLInputElement, Props>(
...
       ...rest
     } = props
+
+    usePropDeprecationWarning({
+      props,
+      name: 'error',
+      componentName: 'Autocomplete',
+      description:
+        'Use the validateStatus prop instead. error is deprecated and will be removed in the next major release.'
+    })
```
Comment (dmaklygin) [inline packages/picasso/src/Autocomplete/Autocomplete.tsx:174]: 👍

File: packages/picasso/src/Input/Input.tsx (hunk 9 above)
Comment (dmaklygin) [reply packages/picasso/src/Input/Input.tsx:257]: I have a question about the usage of different adornments in a row:
I can see `icon` adornment above as well as `LimitAdornment`. Does it mean if we have a limit assigned on the Input component we are not going to show validated adornment?

File: packages/picasso/src/Input/Input.tsx [hunk 14]
```diff
@@ -365,6 +392,7 @@ export const Input = forwardRef<HTMLInputElement, Props>(function Input(
             multiline={multiline}
             counter={counter}
             testIds={testIds}
+            validateStatus={validateStatus}
```
Comment (dmaklygin) [inline packages/picasso/src/Input/Input.tsx:395]: shouldn't we also take into account `error` prop here as for other Components?

File: packages/picasso/src/Input/test.tsx [hunk 15]
```diff
@@ -129,4 +129,16 @@ describe('Input', () => {
     fireEvent.click(inputWrapper!)
     expect(handleClick).toHaveBeenCalledTimes(2)
   })
+
+  it('shows check icon if validateStatus equals to success', () => {
```
Comment (dmaklygin) [inline packages/picasso/src/Input/test.tsx:133]: let's wrap it with `describe` block:
```

describe('when validateStatus equals to success', () => {
  it('shows check icon', () => {
    
  })
})
describe('when validateStatus does not equal to success', () => {
 ...
})
```

File: packages/picasso/src/NumberInput/test.tsx [hunk 16]
```diff
@@ -124,5 +126,17 @@ describe('NumberInput', () => {
 
       expect(input.value).toBe('-95')
     })
+
+    it('shows check icon if validateStatus equals to success', () => {
```
Comment (dmaklygin) [inline packages/picasso/src/NumberInput/test.tsx:130]: I would also wrap it with describe block, as described in the [DOC](https://toptal-core.atlassian.net/wiki/spaces/ENG/pages/1120830304/Frontend+Testing+-+Unit+testing#Describe)

File: packages/picasso/src/NumberInputEndAdornment/NumberInputEndAdornment.tsx [hunk 17]
```diff
@@ -130,6 +138,13 @@ export const NumberInputEndAdornment = (props: Props) => {
 
   return (
     <InputAdornment position='end'>
+      {validateStatus === 'success' && (
+        <CheckMinor24
+          color='green'
+          className={classes.checkMinorIcon}
+          data-testid={testIds?.validIcon}
+        />
+      )}
```
Comment (dmaklygin) [inline packages/picasso/src/NumberInputEndAdornment/NumberInputEndAdornment.tsx:147]: 👍

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx (hunk 2 above)
Comment (ozgurkececioglu) [reply packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:69]: Ok, I thought it would be used only in one place so it looked clear to me, but it actually doesn't. I will change the return type and the function's return statements, and it will solve everything. Do you think it is better to define the type explicitly?

```
const getInputValidateStatus = <T extends ValueType>(
  meta: FieldMeta<T>,
  formConfig: FormConfigProps,
  showValidState?: boolean
): { errorMessage?: string; validateStatus?: ValidateStatus } => {
```

File: packages/picasso/src/PasswordInput/test.tsx [hunk 18]
```diff
@@ -30,4 +26,16 @@ describe('PasswordInput', () => {
 
     expect(input.type).toBe('text')
   })
+
+  it('shows check icon if validateStatus equals to success', () => {
```
Comment (dmaklygin) [inline packages/picasso/src/PasswordInput/test.tsx:30]: let's also wrap it with describe block

File: packages/picasso/src/TagSelector/TagSelector.tsx [hunk 19]
```diff
@@ -80,6 +86,9 @@ export interface Props
   }) => ReactNode
   /** DOM element that wraps the Popper */
   popperContainer?: HTMLElement
+  testIds?: {
+    validIcon?: string
+  }
```
Comment (dmaklygin) [inline packages/picasso/src/TagSelector/TagSelector.tsx:91]: 👍

File: packages/picasso/src/TagSelector/TagSelector.tsx [hunk 20]
```diff
@@ -182,9 +202,17 @@ export const TagSelector = forwardRef<HTMLInputElement, Props>(
       )
     }
 
+    const endAdornment =
+      validateStatus === 'success' ? (
+        <InputAdornment position='end'>
+          <CheckMinor24 color='green' data-testid={testIds?.validIcon} />
+        </InputAdornment>
+      ) : null
+
```
Comment (dmaklygin) [inline packages/picasso/src/TagSelector/TagSelector.tsx:211]: 👍

File: packages/picasso/src/TagSelector/story/ValidateStatus.example.tsx [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+          setInputValue(newInputValue)
+          setOptions(filterOptions(newInputValue))
+        }}
+        validateStatus={value.length === 2 ? 'success' : 'error'}
+      />
+    </Container>
+  )
+}
+
+export default Example
```
Comment (dmaklygin) [inline packages/picasso/src/TagSelector/story/ValidateStatus.example.tsx:74]: 👍

File: packages/picasso/src/TagSelector/test.tsx [hunk 22]
```diff
@@ -124,6 +101,18 @@ describe('TagSelector', () => {
...
+    const { getByTestId } = renderTagSelector({
+      value: [testOptions[0]],
+      validateStatus: 'success',
+      testIds: { validIcon: 'valid-icon' }
+    })
+
+    const validIcon = getByTestId('valid-icon')
+
+    expect(validIcon).toBeVisible()
+  })
```
Comment (dmaklygin) [inline packages/picasso/src/TagSelector/test.tsx:115]: I would also add a second case, when validateStatus isn't `success`

File: packages/picasso/src/Input/Input.tsx (hunk 14 above)
Comment (ozgurkececioglu) [reply packages/picasso/src/Input/Input.tsx:395]: But we will use this prop value just for comparing to success to show the green-check icon. So, the error state is not being used in the `<EndAdornment>`

File: cypress/component/NumberInput.spec.tsx [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+    mount(<NumberInputExample />)
+
+    cy.get('body').happoScreenshot()
+  })
+
+  describe('when validateStatus equals to success', () => {
+    it('shows valid icon', () => {
+      mount(<NumberInputExample validateStatus='success' />)
+
+      cy.get('[data-testid="valid-icon"]').should('be.visible')
```
Comment (dmaklygin) [inline cypress/component/NumberInput.spec.tsx:29]: I guess, this check on `valid-icon` existence was in Unit tests. WDYT to repeat it here as well?

**Review by dmaklygin — COMMENTED** [review]
Good work! Left several comments

File: packages/picasso/src/NumberInput/NumberInput.tsx [hunk 24]
```diff
@@ -73,6 +95,8 @@ export const NumberInput = forwardRef<HTMLInputElement, Props>(
         disabled={disabled}
         size={size}
         inputRef={inputRef}
+        validateStatus={validateStatus}
```
[the remark this answers is not in the document]
Comment (ozgurkececioglu) [reply packages/picasso/src/NumberInput/NumberInput.tsx:98]: Again, error value is not being used inside `NumberInputEndAdornment`. It will be used for just comparing it with `success`

File: cypress/component/NumberInput.spec.tsx (hunk 23 above)
Comment (ozgurkececioglu) [reply cypress/component/NumberInput.spec.tsx:29]: I thought that extra caution would not harm 🙂 I will remove to not populate cypress tests for the future 👍

File: .changeset/afraid-sloths-suffer.md [hunk 25]
```diff
@@ -0,0 +1,9 @@
+---
+'@toptal/picasso-forms': minor
+---
+
+---
+
+### FieldWrapper
```
Comment (denieler) [inline .changeset/afraid-sloths-suffer.md:7]: `FieldWrapper` is not a public component for Picasso users, in changelog it's better to show changes for components which are mentioned in Picasso storybook (sometimes it's better to duplicate the message for each such a component)

File: cypress/component/Select.spec.tsx [hunk 26]
```diff
@@ -380,8 +380,8 @@ describe('Select', () => {
         <div style={{ background: palette.yellow.main }}>
           <TestSelect />
           <TestSelect native />
-          <TestSelect error />
-          <TestSelect native error />
+          <TestSelect validateStatus='error' />
```
Comment (denieler) [inline cypress/component/Select.spec.tsx:383]: maybe just `status` instead of `validateStatus`? there is a similar prop in ant.design - https://ant.design/components/input

Screenshot 2022-03-14 at 13 56 21
/

Comment (ozgurkececioglu) [reply cypress/component/Select.spec.tsx:383]: I have just followed the ticket information. I thought that you guys already talked about this and finalize your variable/prop names. In this case, I can also change the prop for FieldWrapper like `showValid` rather than `showValidState`. WDYT?

File: cypress/component/TagSelector.spec.tsx [hunk 27]
```diff
@@ -59,4 +76,12 @@ describe('TagSelector', () => {
     getOptions().should('have.length', 1)
     getOption('Option 1').should('not.exist')
   })
+
+  describe('when validateStatus equals to success', () => {
```
Comment (denieler) [inline cypress/component/TagSelector.spec.tsx:80]:
```suggestion
  describe('when in a valid state', () => {
```

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:65]: what about having 2 different functions? one for error, another for status? then it might be more granular

File: packages/picasso-forms/src/Form/story/CustomFormLevelConfiguration.example.tsx [hunk 28]
```diff
@@ -3,7 +3,8 @@ import { Container } from '@toptal/picasso'
 import { Form, FormConfigProps } from '@toptal/picasso-forms'
 
 const formConfig: FormConfigProps = {
-  requiredVariant: 'asterisk'
+  requiredVariant: 'asterisk',
+  showValidState: true
```
Comment (denieler) [inline packages/picasso-forms/src/Form/story/CustomFormLevelConfiguration.example.tsx:7]: can we have 2 types of forms? with valid state and without? it's it against BASE?

Comment (TomasSlama) [reply packages/picasso-forms/src/Form/story/CustomFormLevelConfiguration.example.tsx:7]: @denieler yes, BASE team is currently working on preparing documentation, in what cases showcase valid state, when not.

File: packages/picasso-forms/src/Form/story/ValidateStatus.example.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (denieler) [reply packages/picasso-forms/src/Form/story/ValidateStatus.example.tsx:37]: `showValidState` doesn't feel like a field-level configuration. I think it's impossible to have 1 field in a form with valid state, but another without. The form would be felt inconsistent.

[the remark this answers is not in the document]
Comment (ozgurkececioglu) [reply packages/picasso-forms/src/Form/story/ValidateStatus.example.tsx:37]: It makes sense 👍 Then I will get rid of field-level configuration. It can only be enabled via form-level configuration.

File: packages/picasso/src/DatePicker/__snapshots__/test.tsx.snap [hunk 29]
```diff
@@ -25,6 +25,7 @@ exports[`DatePicker renders 1`] = `
           </svg>
         </div>
         <input
+          aria-invalid="false"
```
Comment (denieler) [inline packages/picasso/src/DatePicker/__snapshots__/test.tsx.snap:28]: why this aria argument has appeared?

File: packages/picasso/src/Input/Input.tsx (hunk 9 above)
Comment (denieler) [reply packages/picasso/src/Input/Input.tsx:260]: I think we should ask BASE designers about this

File: cypress/component/Select.spec.tsx (hunk 26 above)
Comment (denieler) [reply cypress/component/Select.spec.tsx:383]: I wasn't part of such conversation, so probably better to discuss that in our channel

Comment (ozgurkececioglu) [reply cypress/component/Select.spec.tsx:383]: We discussed this in the daily and decided to go with `status`. Because it is self-clear to what it means.

File: packages/picasso/src/DatePicker/__snapshots__/test.tsx.snap (hunk 29 above)
Comment (ozgurkececioglu) [reply packages/picasso/src/DatePicker/__snapshots__/test.tsx.snap:28]: As I mentioned here: https://github.com/toptal/picasso/pull/2510#discussion_r824482496, I made them consistent across the repo. There were some optional error prop values and non-optional ones. So, some of the input components were rendering this attribute for `error=false` and some were not. I thought it is ok to render it for all inputs. Do you have a concern about it?

File: packages/picasso/src/Autocomplete/Autocomplete.tsx (hunk 13 above)
Comment (ozgurkececioglu) [reply packages/picasso/src/Autocomplete/Autocomplete.tsx:174]: I thought about that too. I can prepare something along the way separately until we finally decide to get rid of `error`. WDYT?

File: packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx [hunk 30]
```diff
@@ -88,6 +88,33 @@ const getInputError = <T extends ValueType>(
   return meta.submitError
 }
 
+const getInputStatus = <T extends ValueType>(
+  meta: FieldMeta<T>,
+  formConfig: FormConfigProps
+): ValidateStatus | undefined => {
```
Comment (denieler) [inline packages/picasso-forms/src/FieldWrapper/FieldWrapper.tsx:94]:
```suggestion
): Status | undefined => {
```
maybe?
